### ISSUE TITLE
Single message showing multiple times

### ISSUE THREAD

**Issue by foss-** [anurodhp/Monal#153]
For IRC rooms very often times, a single message is shown multiple times. Sometimes that can mean that the message is shown twice, or as shown in the screenshot, many times. It's very annoying because it makes the communication hard to follow and human unreadable.

Tested with Firefox IRC room using following room address: #firefox%irc.mozilla.org@irc.disroot.org

Reproduce should be as easy as joining that room and setting idle for a while, then looking at tghe messages. Allow a few messages to come in.

Happening on both iOS and macOS:
macOS 10.14.2, Monal 2.3 (110)
iOS 12.1.1, Monal 3.3

![single message showing multiple times](https://user-images.githubusercontent.com/5708172/50150896-ab739a80-02bf-11e9-83ed-3cb52969521d.png)

**Comment by foss-** [comment]
Seems to be solved with Version 2.3 (112), closing.

**Comment by foss-** [comment]
not solved :/ re-opening

**Comment by truenicoco** [comment]
A similar bug also exist on "regular" MUCs (no IRC transport) served by a prosody 0.11.2 server the Monal iOS app. It seems it only concerns retrieved history and not live message. Everything works fine using Gajim and Conversations. I would love to help debugging this as it makes reading the MUC history very complicated, let me know if I can help.

![](https://framapic.org/9MIpRz16IGFp/2HJp6gAcy6fR)

**Comment by Echolon** [comment]
@foss- 

Has it improved within the current release?

**Comment by foss-** [comment]
Have not seen dupe entries for a while now. Closing.

**Comment by Echolon** [comment]
:muscle: Sure! Anu needs more support and should focus on the important stuff!

**Comment by foss-** [comment]
Sorry, have to re-open. This is persisting in a xmpp muc for some (not all) messages. Not sure why it happens for some and not for other messages.

**Comment by foss-** [comment]
Another attempt as no duplicate messages have been around for the recent betas.
